go/new_merkletree/merkle_tree_interface.go

```diff
@@ -0,0 +1,61 @@
+package new_merkletree
+
+import "hash"
+
+type Hash []byte
+
+// NewMerkleTreeInterface is the set of methods which are required to
+// manipulate the contents of a Merkle Hash Tree (that is, add new entries),
+// and to query the tree for inclusion and transparency proofs.
+type NewMerkleTreeInterface interface {
```

Comment:
Are we expecting more implementations to satisfy this interface? If not, skip it.